controllers/algae_shooter.py

```diff
@@ -26,8 +26,14 @@ def shoot(self) -> None:
         self.engage()
 
     @state(first=True, must_finish=True)
-    def preparing(self):
+    def preparing(self, initial_call: bool):
         if self.use_ballistics:
+            if initial_call and (
+                not self.ballistics_component.is_in_range()
+                or not self.ballistics_component.is_aligned()
+            ):
```

Comment:
I'd recommend using De Morgan's law here to make this clearer. We want to guard this so when this runs, we are in range and aligned. I think this is easier to reason about.
```suggestion
            if initial_call and not (
                self.ballistics_component.is_in_range()
                and self.ballistics_component.is_aligned()
            ):
```